### DISCUSSION TITLE
Knowing if a message is delivered  to the receiver

### DISCUSSION THREAD

**Discussion by mkassir95** [processone/ejabberd#3976] [Questions] [answered]
Hello,

How to know that  a message is   successfully delivered  to the receiver?

**Comment by badlop** [comment]
There may be some XEP for that. Did you check that? Or do you expect ejabberd people to solve all **your** problems, and do **your** work?

This is not an issue with ejabberd, it seems a general discussion about XMPP. I'll move it to GitHub Discussions.

**Comment by badlop** [comment, marked as answer]
Looking at the list of XEPs, there are two that may help you:
- [XEP-0022: Message Events](https://xmpp.org/extensions/xep-0022.html)
- [XEP-0184: Message Delivery Receipts](https://xmpp.org/extensions/xep-0184.html)

Some clients you can try:
- [Gajim supports XEP-0184](https://gajim.org/support/extensions/)
- [Psi supports both XEPs](https://github.com/psi-im/psi/wiki/Supported-XEPs)
- [ConverseJS supports XEP-0184](https://github.com/conversejs/converse.js#supported-xmpp-extensions)
